src/server/server_test.go

```diff
@@ -0,0 +1,376 @@
+
+	s := busstation.managerHandlers(mux.NewRouter())
+	server := httptest.NewServer(s)
+	defer server.Close()
+	e := httpexpect.New(t, server.URL)
+
+	routes := []domain.Route{
+		{
+			ID: 1,
+			Points: domain.Points{
+				StartPoint: "Vitebsk",
+				EndPoint:   "Minsk",
+			},
+			Start:     time.Date(2019, 04, 23, 10, 0, 0, 0, time.UTC),
+			Cost:      1000,
+			FreeSeats: 12,
+			AllSeats:  13,
+		},
+	}
+
+	testCases := []struct {
+		name           string
+		routeID        int
+		paramID        string
+		expectedStatus int
+		expectedRoute  *domain.Route
+		expectedError  error
+	}{
+		{
+			name:           "successful test",
+			routeID:        1,
+			paramID:        "1",
+			expectedStatus: http.StatusOK,
+			expectedRoute:  &routes[0],
+			expectedError:  nil,
+		},
+		{
+			name:           "no route",
+			routeID:        2,
+			paramID:        "2",
+			expectedStatus: http.StatusInternalServerError,
+			expectedRoute:  nil,
+			expectedError:  errors.New("no such route"),
+		},
+		{
+			name:           "invalid id",
+			paramID:        "df2",
+			expectedStatus: http.StatusBadRequest,
+			expectedRoute:  nil,
+			expectedError:  errors.New("no such route"),
+		},
+	}
+	for _, tc := range testCases {
+		routestrg.On("RouteByID", tc.routeID).Return(tc.expectedRoute, tc.expectedError)
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			res := e.Request(http.MethodGet, "/routes/"+tc.paramID).Expect()
+			res.Status(tc.expectedStatus)
+		})
+	}
+}
+
+func TestCreateRoute(t *testing.T) {
+	cfg := &config.Config{
+		PortServer: 8000,
+	}
+	var routestrg mocks.RouteStorage
+	routeman := routemanager.NewRouteManager(&routestrg)
+	busstation := NewBusStation(routeman, cfg)
+
+	s := busstation.managerHandlers(mux.NewRouter())
+	server := httptest.NewServer(s)
+	defer server.Close()
+	e := httpexpect.New(t, server.URL)
+
+	routes := []domain.Route{
+		{
+			Points: domain.Points{
+				StartPoint: "Vitebsk",
+				EndPoint:   "Minsk",
+			},
+			Start:     time.Date(2002, 04, 23, 10, 0, 0, 0, time.UTC),
+			Cost:      1000,
+			FreeSeats: 12,
+			AllSeats:  13,
+		},
+		{
+			Points: domain.Points{
+				StartPoint: "Grodno",
+				EndPoint:   "Minsk",
+			},
+			Start:     time.Date(2021, 04, 12, 10, 0, 0, 0, time.UTC),
+			Cost:      1000,
+			FreeSeats: 12,
+			AllSeats:  13,
+		},
+	}
+	testCases := []struct {
+		name           string
+		route          domain.Route
+		expectedStatus int
+		expectedID     int
+		expectedError  error
+	}{
+		{
+			name:           "errors",
+			route:          routes[0],
+			expectedStatus: http.StatusBadRequest,
+			expectedID:     1,
+			expectedError:  errors.New("date is invalid"),
+		},
+		{
+			name:           "successful test",
+			route:          routes[1],
+			expectedStatus: http.StatusOK,
+			expectedID:     1,
+			expectedError:  nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		routestrg.On("AddRoute",
+			tc.route.Points.StartPoint,
```

Comment:
You can notice, that here there are a lot of params for function AddRoute. You can rewrite it as
`AddRoute(r route)` in the db